x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/oidc/OPConfiguration.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+package org.elasticsearch.xpack.security.authc.oidc;
+
+import org.elasticsearch.common.Nullable;
+
+import java.util.Objects;
+
+/**
+ * A Class that contains all the OpenID Connect Provider configuration
+ */
+public class OPConfiguration {
```

Comment:
I wonder if we should define `OP` and `RP` somewhere? It requires some pretty specific knowledge to know these acronyms and we've been spelling out vs abbreviating for things like `ActiveDirectory`. Something to think about.